locale.c

```diff
@@ -1683,16 +1683,14 @@ S_calculate_LC_ALL(pTHX_ const char ** individ_locales)
         names_len += strlen(category_names[i])
                   + 1                           /* '=' */
                   + strlen(entry)
-                  + 1;                          /* ';' */
+                  + 1;                          /* ';' or terminating NUL */
     }
 
-    names_len++;    /* Trailing '\0' */
-
     /* Allocate enough space for the aggregated string */
     SAVEFREEPV(Newxz(aggregate_locale, names_len, char));
 
     /* Then fill it in */
-    for (i = 0; i < NOMINAL_LC_ALL_INDEX; i++) {
+    for (unsigned i = 0; i < NOMINAL_LC_ALL_INDEX; i++) {
```

Comment:
I know it is legal, but could we spell this out fully? it makes me a bit leery to see a raw unsigned, even if it is legal.